internal/resources/cluster/nodepools/resource_node_pool.go

```diff
@@ -24,6 +24,8 @@ const (
 	ResourceName = "tanzu-mission-control_cluster_node_pool"
 )
 
+type contextMethodKey struct{}
```

Comment:
let's not duplicate the code and have the context creation in a lib and resued.
and in the same lib, have functions for creating a context with function name

Reply:
Please check the changes for nodepool resource, I will add the same to others once you approve these.

Reply:
Approved